### DISCUSSION TITLE
SpanId and traceId always zeroes

### DISCUSSION THREAD

**Discussion by ghost** [open-telemetry/opentelemetry-java-instrumentation#9475] [Q&A] [answered]
Hello
We have very strange behaviour. 
adding: 
`        <dependency>
            <groupId>io.micrometer</groupId>
            <artifactId>micrometer-tracing-bridge-otel</artifactId>
        </dependency>

        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-exporter-otlp</artifactId>
        </dependency>`
will generate valid trace id and span id, but only for the system logs, everything that is logged within application itself do not contains.
"traceId":"cbfc93cd4569d89aaee0788aaa94b7e4","spanId":"2049c5706d9e0602"
however logging section yaml in spring boot looks like this:
`
logging:
  pattern:
    level: ${spring.application.name:}, trace_id=%mdc{trace_id} span_id=%mdc{span_id} trace_flags=%mdc{trace_flags} %5p
  level:
    reactor:
      netty: INFO
    org:
      springframework:
        cloud:
          gateway: INFO
`
As you can see within pattern we are using 'trace_id' not 'traceId'
And finally String traceId = Span.current().getSpanContext().getTraceId(); will be '00000000000000000000000000000000'
Following manual implementation.
Added:

    @Bean
    public SdkTracerProvider sdkTracerProvider(@Autowired Resource resource) {
        SdkTracerProvider sdkTracerProvider = SdkTracerProvider.builder()
                .addSpanProcessor(SimpleSpanProcessor.create(LoggingSpanExporter.create()))
                .setResource(resource)
                .build();
        return sdkTracerProvider;
    }

    @Bean
    public SdkMeterProvider sdkMeterProvider(@Autowired Resource resource) {
        SdkMeterProvider sdkMeterProvider = SdkMeterProvider.builder()
                .registerMetricReader(PeriodicMetricReader.builder(LoggingMetricExporter.create()).build())
                .setResource(resource)
                .build();
        return sdkMeterProvider;
    }

    @Bean
    public SdkLoggerProvider sdkLoggerProvider(@Autowired Resource resource) {
        SdkLoggerProvider sdkLoggerProvider = SdkLoggerProvider.builder()
                .addLogRecordProcessor(BatchLogRecordProcessor.builder(SystemOutLogRecordExporter.create()).build())
                .setResource(resource)
                .build();
        return sdkLoggerProvider;
    }

    @Bean
    public OpenTelemetry openTelemetry(@Autowired SdkTracerProvider sdkTracerProvider,
                                       @Autowired SdkMeterProvider sdkMeterProvider,
                                       @Autowired SdkLoggerProvider sdkLoggerProvider) {
        OpenTelemetry openTelemetry = OpenTelemetrySdk.builder()
                .setTracerProvider(sdkTracerProvider)
                .setMeterProvider(sdkMeterProvider)
                .setLoggerProvider(sdkLoggerProvider)
                .setPropagators(ContextPropagators.create(TextMapPropagator.composite(W3CTraceContextPropagator.getInstance(),
                        W3CBaggagePropagator.getInstance())))
                .build();
        return openTelemetry;
    }

    @Bean
    public OtlpHttpSpanExporter otlpHttpSpanExporter(@Value("${tracing.url}") String url) {
        return OtlpHttpSpanExporter.builder().setEndpoint(url).build();
    }
     @Bean
      public Resource resource(@Value("${spring.application.name}") String applicationName) {
          return Resource.getDefault().toBuilder().put(ResourceAttributes.SERVICE_NAME, 
           applicationName).put(ResourceAttributes.SERVICE_VERSION, "0.1.0").build();
      }
`
And finally: 
`        <dependency>
            <groupId>io.micrometer</groupId>
            <artifactId>micrometer-tracing-bridge-otel</artifactId>
        </dependency>

        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-exporter-otlp</artifactId>
        </dependency>

        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-api</artifactId>
        </dependency>
        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-sdk</artifactId>
        </dependency>
        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-sdk-metrics</artifactId>
        </dependency>
        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-exporter-logging</artifactId>
        </dependency>`
Have no clue what I am doing wrong. Debugging couple of days already

**Comment by trask** [comment, marked as answer]
hi @maxim-badarau-m10! I'd recommend opening issue in `micrometer-tracing-bridge-otel` for help with that library.

If you are looking for something similar in this repository, I'd recommend https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/spring/spring-boot-autoconfigure (or possibly the opentelemetry java agent if you just want the easiest way to onboard to opentelemetry)
